### PR TITLE
Refactor Dashboard code

### LINKED ISSUE #1876: Refactor dashboard code
Break the dashboard code into submodules. At the same time, ensure that the charts will work with multiproxy.

# Definition of done
* [ ] Instance functions are moved to own js files
* [ ] Use ES import for chart functions
* [ ] Make sure all charts work and are still interactive

# Notes from exploratory session
List of things to refactor in Dashboard code:

# Current structure
The Dashboard code has the following structure:

- client/
  - charts/
    - bar/ ...
    - filtering/ ...
    - line/ ...
    - overview/ ...
    - row/ ...
    - statistics/ ...
    - table/ ...
    - charts.html
    - charts.js
    - charts.less
  - lib/
  - dashboard.html
  - dashboard.js
  - dashboard.less

## Complexity
The main areas of complexity are within `charts.js` and `dashboard.js`.

### charts.js
What is in the charts.js file? What does it do?

The charts.js file manages

- chartData
- currentData.loadingState
- api frontend prefixes
- filtered chart data
- add/remove loading placeholders (DOM)
- parsed chart data
- rendering charts
- table data

#### Methods
The dashboardCharts template has the following instance **methods**:

- parseChartData
- renderCharts
- getTableData
- updateDataTable
- updateLineChart
- filterData
- updateStatisticsData
- getRequestsCount
- getAverageResponseTime
- getResponseRate
- getUniqueUsersCount

Some of these template methods could be moved into the templates where they are directly used. Then, the 'chartData' would just be passed directly into each chart component, or we could even make a request specifically for the chart component.

#### Event handlers
The dashboardCharts has the following **event handlers**:

- 'change api-frontend-prefix-form' : sets a list of frontend prefixes from UI widget
- 'click tick-hour' - updates a CSS class, sets the timestamp format for analytics data
- 'click tick-day' - sets or removes a CSS class, sets timestamp format for analytics data
- 'click tick-week' - sets or removes a CSS class, sets timestamp format for analytics data
- 'click tick-month' - sets or removes a CSS class, sets timestamp format for analytics data

It may be possible to  combine, or even remove, the tick (day, week, month) handlers and move the result methods to `tickButtons` template

### dashboard.js
What is in the dashboard.js file? What does it do?

#### Methods
The dashboard template has the following instance **methods**:

- checkElasticsearch
- getChartData
- getElasticSearchQuery

# Desired structure
We would like to structure the dashboard code in a simpler manner, such as:

- client/
  - bar/ ...
  - filtering/ ...
  - line/ ...
  - overview/ ...
  - row/ ...
  - statistics/ ...
  - table/ ...
- server/

### LINKED ISSUE #1904: Dashboard view: displays 'No Data Found' for selected dates while data available
# Steps 
1. Visit https://staging.apinf.io 
2. Sign up  
3. Add API backend with proxy    
4. Call API 
5. Go to 'Dashboard'  
6. Select the above API added from the drop down menu
7. Select the dates in calendar - 'from' and 'to'   (eg. 1 Nov 2016(any prior date) - 9 Nov 2016 (current day)) 

# Expected Result
Displays data for selected dates  

# Actual Result  
Displays a message 'No Data Found' while data is available   

# Browsers 
-  Google chrome version 54.0.2840.87 m
-  Internet Explorer 11  

# Operating system  
Windows 10    

# Screenshot 
screenshot 38

### PR DESCRIPTION
**Pull request by brylie**
Closes #1876 
Closes #1904

# Changes
- refactor state handling for proxy api backends check
- edit .dropdown-toggle font color (API Select List)
- refactor filter method to work with single proxy frontent prefix
- remove imperative 'loading' state UI management
- move start and end date state to URL

# Known issues
- Loading spinners have been disabled and need to be re-added (enhancement?)
- API Select does not trigger analytics reload
- Dashboard 'no proxy added' text is behaving oddly

### CODE REVIEW

**Comment by jykae** [comment]
@brylie taking a look

**Comment by jykae** [comment]
@brylie nice work overall so far 👍

**Comment by marla-singer** [comment]
> API Select does not trigger analytics reload

I think it happens because we don't get data for selected API. Looks like we gather the information about all proxy backend and don't orientate to selcted API.
https://github.com/apinf/platform/blob/develop/dashboard/client/dashboard.js#L69-L90
I can be wrong, it's just supposition

File: dashboard/client/dashboard.js [hunk 1]
```diff
@@ -3,53 +3,21 @@ import { Template } from 'meteor/templating';
...
-
-  // Keeps date format for moment.js
-  instance.dateFormatMoment = 'DD MMM YYYY';
-
-  // Init default time frame (from: 2weeks ago, to: now)
-  instance.analyticsTimeframeStart = new ReactiveVar(moment().subtract(1, 'month'));
-  instance.analyticsTimeframeEnd = new ReactiveVar(moment());
-
-  // Get current user Id
-  const userId = Meteor.userId();
```
Comment (marla-singer) [inline dashboard/client/dashboard.js]: Be careful. Eslint promts that userId isn't defined in this line https://github.com/apinf/platform/blob/c758fbf7c0b97822da7c72c6f574ef5c683f5ccf/dashboard/client/dashboard.js#L61

**Comment by marla-singer** [comment]
@brylie Have error in browser consoleon dashboard page:
`Cannot read property 'currentUserIsManager' of undefined`
Fail is here: https://github.com/apinf/platform/blob/develop/proxy_backends/collection/helpers.js#L11
apiId is correct but api is undefined
This helper is used here https://github.com/apinf/platform/blob/c758fbf7c0b97822da7c72c6f574ef5c683f5ccf/dashboard/client/filtering/filtering.js#L35

**Comment by bajiat** [comment]
@marla-singer Can you review this PR? Reviewing this would probably help you in your task.

**Comment by marla-singer** [comment]
@bajiat Of course, anyway I'm goint to helpwith review to Ville

**Pull request closed without merging**